### PR TITLE
Moves HTML anchor into header and creates valid HTML

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

HTML files are created from markdown files. The current markdown code puts the anchor in a paragraph before the headline the link should link to. The solution does not create valid HTML. This PR resolves the issue.

## Changes

All FAQ files in `docs/FAQ/gui-faq-*.md`, but no change of content.

## How to test this PR

Links in table of contents should go to the right question.

### CODE REVIEW

**Review by marc-vanderwal — DISMISSED** [review]
Looks good to me.

**Comment by blacksponge** [comment]
Please update the translation guide https://github.com/zonemaster/zonemaster-gui/blob/master/docs/TranslationGuide.md#gui-faq-langmd

And the style of the FAQ https://github.com/zonemaster/zonemaster-gui/blob/master/src/app/components/faq/faq.component.css#L1

**Comment by blacksponge** [comment]
`faq.component.css` needs to be updated as well, specifically the `ngAfterViewChecked` method,

```js
  ngAfterViewChecked(): void {
    if (!this.fragment) {
      return
    }

    const faqEntry = document.getElementById(this.fragment);
    if (faqEntry !== null) {
      faqEntry.scrollIntoView();
      this.fragment = '';
    }
  }
```

While testing I noticed a strange behaviour when clicking on a link for the first time on Firefox, the browser does not  scroll down all the way to the FAQ entry but stops with a warning in the console saying `Scroll anchoring was disabled in a scroll container because of too many consecutive adjustments`. This seems to be related to (scroll anchoring)[https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor/Guide_to_scroll_anchoring].  Adding the following into the component style fixes the issue.
```css
.faq {
  overflow-anchor: none;
}
```

**Comment by matsduf** [comment]
@blacksponge  and @rhedman, as far as I understand, the old content of src/app/components/faq/faq.component.css is not needed anymore.

**Comment by blacksponge** [comment]
> as far as I understand, the old content of src/app/components/faq/faq.component.css is not needed anymore.

It is, the selector should be updated to `span[id^=q] ` instead of `a[name^=q] `.
This rule is to offset the scroll to avoid having the entry title hidden by the navigation menu at the top.

File: src/app/components/faq/faq.component.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, AfterViewChecked, OnDestroy, Inject, LOCALE_ID } from '@angular/core';
+Aimport { Component, OnInit, AfterViewChecked, OnDestroy, Inject, LOCALE_ID } from '@angular/core';
```
Comment (blacksponge) [inline src/app/components/faq/faq.component.ts:1]: Extra `A`.

**Comment by matsduf** [comment]
> > as far as I understand, the old content of src/app/components/faq/faq.component.css is not needed anymore.
> 
> It is, the selector should be updated to `span[id^=q] ` instead of `a[name^=q] `. This rule is to offset the scroll to avoid having the entry title hidden by the navigation menu at the top.

I will put it back, updated.

File: src/app/components/faq/faq.component.ts (hunk 1 above)
Comment (matsduf) [reply src/app/components/faq/faq.component.ts]: Fixed in conflict resolution.

**Comment by rhedman** [comment]
@blacksponge @matsduf 
Could we create a setting in the config that completely turns off all styles? Something like: `globalStyles: false`. I'm implementing a new design and it's a hassle to always have to write CSS-selectors that overrides the default CSS.

**Comment by matsduf** [comment]
@blacksponge and @rhedman, please rereview.

**Review by rhedman — CHANGES_REQUESTED** [review]
Change this selector: :host ::ng-deep span[id^=q]
Into this .faq span[id^=q]

File: src/app/components/faq/faq.component.css [hunk 2]
```diff
@@ -1,3 +1,8 @@
+:host ::ng-deep span[id^=q] {
```
[the remark this answers is not in the document]
Comment (matsduf) [reply src/app/components/faq/faq.component.css]: I have updated the CSS file.

**Review by rhedman — APPROVED** [review]

**Review by blacksponge — APPROVED** [review]

**Comment by pnax** [comment]
### v2023.1 release testing

Clicking on the links in the FAQ scroll to the right section. The section title is properly displayed at the top of the viewport, just below the header.

**Pull request merged**